libtenzir/builtins/formats/json.cpp

```diff
@@ -195,85 +209,96 @@ class doc_parser {
         value_parse_success
           = parse_value(val.value_unsafe(), builder.field(key), depth + 1);
       }
-      if (not value_parse_success) {
-        return false;
+      written_once = true;
+      if (value_parse_success != result::success) {
+        return result::failure_with_write;
```

Comment:
How do you know that you have written something here if it failed? `.field()`, at least for `series_builder`, is lazy.